src/pages/index.js

```diff
@@ -112,51 +118,223 @@ const modalPreviewCloseButton = document.querySelector("#modalPreviewClose");
 //   document.removeEventListener("click", handleOverlayClose);
 // }
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "23172f33-55e2-4e0e-a695-0bae3ab40106",
+});
+
+let cardSection;
+
+api.getInitialCards().then((cards) => {
+  cardSection = new Section({ items: cards }, renderCard, "#section");
+
+  function renderCard(cards) {
+    const cardElement = new Card(
+      cards,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleCardLike,
+      handleCardLikeRemove
+    );
+    console.log(cards);
+    cardSection.addItem(cardElement.getView());
+  }
+
+  cardSection.renderItems();
+});
+
 const newCardPopup = new PopupWithForm({
   popupSelector: "#contentModal",
   handleFormSubmit: handleCardFormSubmit,
 });
 
+contentAddButton.addEventListener("click", () => {
+  cardFormValidator.toggleButtonState();
+  newCardPopup.openModal();
+});
+
+function handleCardFormSubmit(cardValues) {
+  const submitButton = document.getElementById("modalContentSubmit");
+  const originalButtonText = submitButton.textContent;
+  // Change button text to "Saving..."
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardValues)
+    .then((cardData) => {
+      const newCard = new Card(
+        cardData,
+        "#card-template",
+        handleImageClick,
+        handleCardDeleteClick,
+        handleCardLike,
+        handleCardLikeRemove
+      );
+      cardSection.prependItems(newCard.getView());
+      submitButton.textContent = originalButtonText;
+      newCardPopup.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
+      submitButton.textContent = originalButtonText;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`. No need to duplicate the 2 times

**Please, correct it everywhere**